symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,192 @@
+ *     Eigen::Vector3d GetGyroBiasEstimate(double time);
+ *     void AppendOtherFactors(std::vector<sym::Factord> & factors);
+ *
+ *     // Example accelerometer and gyroscope covariances
+ *     const Eigen::Matrix3d accel_cov = 1e-5 * Eigen::Matrix3d::Identity();
+ *     const Eigen::Matrix3d gyro_cov = 1e-5 * Eigen::Matrix3d::Identity();
+ *
+ *     std::vector<sym::Factord> factors;
+ *
+ *     // GetKeyFrameTimes assumed to return at least 2 times
+ *     std::vector<double> key_frame_times = GetKeyFrameTimes();
+ *
+ *     // Integrating Imu measurements between keyframes, creating an ImuFactor for each
+ *     for (unsigned int i = 0; i < key_frame_times.size() - 1; i++) {
+ *       const double start_time = key_frame_times[i];
+ *       const double end_time = key_frame_times[i + 1];
+ *
+ *       const std::vector<ImuMeasurement> measurements = GetMeasurementsBetween(start_time,
+ *                                                                               end_time);
+ *
+ *       // ImuPreintegrator should be initialized with the best guesses for the IMU biases
+ *       sym::ImuPreintegrator<double> integrator(GetAccelBiasEstimate(start_time),
+ *                                                GetGyroBiasEstimate(start_time));
+ *       for (const ImuMeasurement& meas : measurements) {
+ *         // IntegrateMeasurement assumes IMU measures acceleration of magnitude |gravity| pointing
+ *         // away from the Earth when stationary
+ *         integrator.IntegrateMeasurement(meas.acceleration, meas.acceleration, accel_cov,
+ *                                         gyro_cov, meas.duration);
+ *       }
+ *
+ *       factors.push_back(sym::ImuFactor<double>(integrator)
+ *                             .Factor({{Var::POSE, i},
+ *                                      {Var::VELOCITY, i},
+ *                                      {Var::POSE, i + 1},
+ *                                      {Var::VELOCITY, i + 1},
+ *                                      {Var::ACCEL_BIAS, i},
+ *                                      {Var::GYRO_BIAS, i},
+ *                                      {Var::GRAVITY},
+ *                                      {Var::EPSILON}}));
+ *     }
+ *
+ *     AppendOtherFactors(factors);
+ *
+ *     sym::Optimizerd optimizer(sym::DefaultOptimizerParams(), factors);
+ *
+ *     // Build Values
+ *     sym::Valuesd values;
+ *     for (int i = 0; i < key_frame_times.size(); i++) {
+ *       values.Set({Var::POSE, i}, sym::Pose3d());
+ *       values.Set({Var::VELOCITY, i}, Eigen::Vector3d::Zero());
+ *     }
+ *     for (int i = 0; i < key_frame_times.size() - 1; i++) {
+ *       values.Set({Var::ACCEL_BIAS, i}, GetAccelBiasEstimate(key_frame_times[i]));
+ *       values.Set({Var::GYRO_BIAS, i}, GetGyroBiasEstimate(key_frame_times[i]));
+ *     }
+ *     // gravity should point towards the direction of acceleration
+ *     values.Set({Var::GRAVITY}, (Eigen::Vector3d() << 0.0, 0.0, -9.8).finished());
```

Comment:
This can just be `Eigen::Vector3d(0, 0, -9.8)`

Reply:
Oh cool. I didn't know you could do that.